x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/EsqlTranslatorHandler.java

```diff
@@ -245,4 +258,91 @@ private static boolean isInRange(DataType numericFieldDataType, DataType valueDa
             return minValue.compareTo(decimalValue) <= 0 && maxValue.compareTo(decimalValue) >= 0;
         }
     }
+
+    public static class SpatialRelatesTranslator extends ExpressionTranslator<SpatialRelatesFunction> {
+
+        @Override
+        protected Query asQuery(SpatialRelatesFunction bc, TranslatorHandler handler) {
+            return doTranslate(bc, handler);
+        }
+
+        public static void checkSpatialRelatesFunction(Expression constantExpression, ShapeField.QueryRelation queryRelation) {
+            Check.isTrue(
+                constantExpression.foldable(),
+                "Line {}:{}: Comparisons against fields are not (currently) supported; offender [{}] in [ST_{}]",
+                constantExpression.sourceLocation().getLineNumber(),
+                constantExpression.sourceLocation().getColumnNumber(),
+                Expressions.name(constantExpression),
+                queryRelation
+            );
+        }
+
+        /**
+         * We should normally be using the real `wrapFunctionQuery` above, so we get the benefits of `SingleValueQuery`,
+         * but at the moment `SingleValueQuery` makes use of `SortDocValues` to determine if the results are single or multi-valued,
+         * and LeafShapeFieldData does not support `SortedBinaryDocValues getBytesValues()`.
+         * Skipping this code path entirely is a temporary workaround while separate work is being done to simplify `SingleValueQuery`
+         * to rather rely on a new method on `LeafFieldData`. This is both for the benefit of the spatial queries, as well as an
+         * improvement overall.
+         * TODO: Remove this method and call the parent method once the SingleValueQuery improvements have been made
```

Comment:
Nik's going to shift the guts of `SingleValueQuery` around a bit to make it more OOP so we can make it compatible with this branch. I'll see if I can do this soon, though it might have to be a followup to this PR.